teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java

```diff
@@ -257,10 +292,19 @@ private void initialise() {
             new PublicKeyLoader(),
             asyncRunner,
             metricsSystem,
-            Optional.empty());
+            dataDirLayout);
 
     try {
       validatorLoader.loadValidators();
+      pubKeysToExit.ifPresent(
+          blsPublicKeys ->
+              validatorLoader.getOwnedValidators().getActiveValidators().stream()
```

Comment:
I'd also do something like:
```java
      final OwnedValidators ownedValidators = validatorLoader.getOwnedValidators();

      pubKeysToExit.ifPresent(
          blsPublicKeys ->
                  ownedValidators.getActiveValidators().stream()
                  .filter(validator -> !blsPublicKeys.contains(validator.getPublicKey()))
                  .forEach(
                      validator ->
                              ownedValidators
                              .removeValidator(validator.getPublicKey())));
      validators = ownedValidators;
```

to reduce the interaction with the loader and work with a local owned validators

Reply:
I'd ideally work with a local data structure to avoid calling `OwnedValidators::removeValidator` which is a bit scary if in the future we add some logic that persists something somewhere...

Reply:
I did the opposite: When there is filter on public keys, I pick the concerned validators and add them to the `OwnedValidators` `validators` variable. Plus I used a local data structure as suggested

Reply:
I completely removed the `OwnedValidators` and replaced it by a local `HashMap`. I hope that's what you had in mind